src/Surfnet/Stepup/Configuration/Event/NewInstitutionConfigurationCreatedEvent.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * Copyright 2016 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\Stepup\Configuration\Event;
+
+use Broadway\Serializer\SerializableInterface;
+use Surfnet\Stepup\Configuration\Value\Institution;
+use Surfnet\Stepup\Configuration\Value\InstitutionConfigurationId;
+
+class NewInstitutionConfigurationCreatedEvent implements SerializableInterface
+{
+    /**
+     * @var InstitutionConfigurationId
+     */
+    public $institutionConfigurationId;
+
+    /**
+     * @var Institution
+     */
+    public $institution;
+
+    public function __construct(InstitutionConfigurationId $institutionConfigurationId, Institution $institution)
+    {
+        $this->institutionConfigurationId = $institutionConfigurationId;
+        $this->institution                = $institution;
+    }
+
+    public static function deserialize(array $data)
+    {
+        return new self(
+            new InstitutionConfigurationId($data['institution_configuration_id']),
+            new Institution($data['institution'])
+        );
+    }
+
+    public function serialize()
+    {
+        return [
+            'institution_configuration_id' => (string) $this->institutionConfigurationId,
```

Comment:
We did use string casting on ID value objects near the start of this project, but in time we migrated to explicitly extracting values from ID value objects. A value object may for instance be a compound value (IP address + subnet mask), and you would retrieve these values separately. You can see we've reflected this decision in [`IdentityId`](https://github.com/SURFnet/Stepup-Middleware/blob/b6516e39ec80f667f239d99bfa80b913cc1ccf1b/src/Surfnet/Stepup/Identity/Value/IdentityId.php#L46), for example.

Also, in my opinion `__toString()` is for creating a string representation of the object itself. So for a compound value like an IP address + subnet mask, the string representation would contain both values. Basically `__toString` is very useful for logging etc., but it's not a very explicit manner of communicating with this object.

So for the earlier reasons and consistency's sake, please extract the value explicitly.